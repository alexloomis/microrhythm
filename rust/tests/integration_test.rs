use microrhythm::resolve::*;

fn test_measure() -> Measure {
    Measure {
        mix: 0.5,
        beats: vec![
            Pattern {
                length: 2.0,
                delay: 0.0,
                subdiv: None,
            },
            Pattern {
                length: 1.0,
                delay: -0.2,
                subdiv: Some((
                    0.4,
                    vec![
                        Pattern {
                            length: 2.0,
                            delay: 0.3,
                            subdiv: None,
                        },
                        Pattern {
                            length: 1.0,
                            delay: 0.1,
                            subdiv: Some((
                                0.6,
                                vec![
                                    Pattern {
                                        length: 2.0,
                                        delay: 0.0,
                                        subdiv: None,
                                    },
                                    Pattern {
                                        length: 2.0,
                                        delay: 0.0,
                                        subdiv: None,
                                    },
                                ],
                            )),
                        },
                    ],
                )),
            },
            Pattern {
                length: 1.0,
                delay: 0.0,
                subdiv: None,
            },
        ],
    }
}

#[test]
fn test_resolve() {
    insta::assert_yaml_snapshot!(resolve(test_measure()))
}

