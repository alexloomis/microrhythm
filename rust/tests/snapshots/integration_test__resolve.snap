---
source: tests/integration_test.rs
expression: resolve(test_measure())

---
- 0.34285714285714288
- 0.23214285714285716
- 0.06964285714285715
- 0.06964285714285715
- 0.2857142857142857

